test/Middleware/AuthHttpBasicTest.php

```diff
@@ -25,72 +25,122 @@ protected function getMiddleware()
-     
-        $notAuth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER'); //nicht authorisiert->willreturn(false)
+        //no authorization header
+        $noAuthHeader = $this->recentlyHandledRequest->getAttribute('NO_AUTH_HEADER'); 
+        //NULL = notAuthenticated
+        $notAuth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER'); 
         
-        $this->assertNull($notAuth); //not authenticated
+        $this->assertNull($notAuth); 
         $this->assertEquals($username, $noAuthHeader); 
        
         
       
     }
     
-    public function testNotAuthenticatedWithHeader()  //test ob user nicht authentifiziert
+    public function testNotAuthenticatedWithHeader()  
     {
         $username = 'testUser01';
         
+        //not authenticated
         $this->authDriver->method('authenticate')->willReturn(false);
+
         $this->registry->method('getAuth')->willReturn($username);
-        $request = $this->requestFactory->createServerRequest('GET', '/test');
+        
+        
+        //with authorization header 
+        $request = $this->requestFactory->createServerRequest('GET', '/test')->withHeader('Authorization','BASIC YXNkZjphc2QxMjM=');
         $middleware = $this->getMiddleware();
         $response = $middleware->process($request, $this->handler);
         
-        $AuthHeader = !$this->recentlyHandledRequest->getAttribute('NO_AUTH_HEADER'); //mit authorization header
-        $notAuth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER');
+        $authHeader = $this->recentlyHandledRequest->hasHeader('Authorization');
 
-        $this->assertFalse($AuthHeader); //not no_auth_header
+        $notAuth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER');
         
-        $this->assertNull($notAuth); //not authenticated
+        
+        //header available
+        $this->assertTrue($authHeader);  
+        
+        //NULL
+        $this->assertNull($notAuth);      
         
         
-      
     }
    
-    public function testAuthenticated()  //test ob user authentifiziert
+    public function testAuthenticatedwithoutHeader()  //test ob user authentifiziert
     {
         $username = 'testUser01';
+
         
-         
+     
+         //authenticated USER AND PW accepted
         $this->authDriver->method('authenticate')->willReturn(true);
+        
         $this->registry->method('getAuth')->willReturn($username);
+
+        //check for header
         $request = $this->requestFactory->createServerRequest('GET', '/test');
         $middleware = $this->getMiddleware();
         $response = $middleware->process($request, $this->handler);
+
+        $noAuthHeader = $this->recentlyHandledRequest->getAttribute('NO_AUTH_HEADER'); 
+        $auth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER',$username); 
```

Comment:
This is wrong. If you provide a fallback here it will always return the username. You want to call this without the `$username` fallback and then check that it returned `null`.